src/source/tile.js

```diff
@@ -418,7 +419,7 @@ class Tile {
         }
     }
 
-    setFeatureState(states: LayerFeatureStates) {
+    setFeatureState(states: LayerFeatureStates, painter: any) {
```

Comment:
I used `painter` to maintain similarity with `loadVectorData`, but it doesn't have to be that way - for both these methods. The only advantage I see of passing `painter` as the argument is that it communicates that the _render style_ is being used. This becomes important when considering the Immutable Styl/Render-classes split.

Reply:
Oh, yeah that's odd that `loadVectorData` takes a painter rather than a style object.  I lean slightly towards passing `Style` here, but since this will get refactored in the nearish future with the immutable style state changes, it's not a big deal either way.  

> The only advantage I see of passing painter as the argument is that it communicates that the render style is being used. This becomes important when considering the Immutable Styl/Render-classes split

I think the correct style to use here will be the one that the `tile` will already hold a reference to -- namely, the one that was used at layout time, since that's the one that will be consistent with the data in the buckets.

Reply:
>  since this will get refactored in the nearish future with the immutable style state changes, it's not a big deal either way.

I'd prefer to keep it as-is in that case.

> I think the correct style to use here will be the one that the tile will already hold a reference to

👍